src/loops.rs

```diff
@@ -228,6 +231,17 @@ impl LateLintPass for LoopsPass {
                 }
             }
         }
+        if let ExprMatch(ref expr, ref arms, MatchSource::WhileLetDesugar) = expr.node {
+            let pat = &arms[0].pats[0].node;
+            if let (&PatEnum(ref path, _), &ExprMethodCall(method_name, _, _)) = (pat, &expr.node) {
+                if method_name.node.as_str() == "next" &&
+                        match_trait_method(cx, expr, &["core", "iter", "Iterator"]) &&
+                        path.segments.last().unwrap().identifier.name.as_str() == "Some" {
```

Comment:
Expect is fine.

Note that when a plugin crashes, it looks like an ICE, so really we should be using `span_bug` or something here with an if let.

(Also, the CI won't accept a clippy PR which emits clippy warnings, run `bash util/dogfood.sh` to see)